### ISSUE TITLE
PID after fork is fixed

### ISSUE THREAD

**Issue by fcharlie** [gabime/spdlog#385]
**White content** is child process output 

![2017-03-13 17-12-12](https://cloud.githubusercontent.com/assets/6904176/23848046/3b773490-0810-11e7-9b1f-04c4f35a80b0.png)

TID  also is fixed after fork()

spdlog commit id: 270c08b275e8c2f58264932b45f0f27bc3169ee0

**Comment by asit-dhal** [comment]
Thread id is expected to be unique inside a process, two thread ids of two separate processes may be equal.

Is this reproducible ?

So, in order to experiment this, I wrote a sample examples and it also showed similar results. The thread id comes from the standard library. As far as my understanding, every time an msg is created, thread id is received from the library(I am not sure). May be @gabime  can confirm this.

Here is the sample code 
https://github.com/asit-dhal/spdlog/blob/issue_385/example/example_mt.cpp
https://github.com/asit-dhal/spdlog/blob/issue_385/example/example_st.cpp

The output
```

*** [00:03:42 +02:00] [parent 23477] Parent Master Thread ***
*** [00:03:42 +02:00] [parent 23482] Easy padding in numbers like 00000012 ***
*** [00:03:42 +02:00] [parent 23482] Positional args are supported too.. ***
*** [00:03:42 +02:00] [parent 23481] Easy padding in numbers like 00000012 ***
*** [00:03:42 +02:00] [parent 23481] Positional args are supported too.. ***
*** [00:03:42 +02:00] [parent 23480] Easy padding in numbers like 00000012 ***
*** [00:03:42 +02:00] [parent 23480] Positional args are supported too.. ***
*** [00:03:42 +02:00] [parent 23479] Easy padding in numbers like 00000012 ***
*** [00:03:42 +02:00] [parent 23479] Positional args are supported too.. ***
*** [00:03:42 +02:00] [parent 23477] End of example. tid=140392174827392 ***
*** [00:03:42 +02:00] [child 23478] Child Master Thread ***                                                                                                                                          
*** [00:03:42 +02:00] [child 23487] Welcome to spdlog! ***
*** [00:03:42 +02:00] [child 23487] Support for floats 1.23 ***
*** [00:03:42 +02:00] [child 23486] Welcome to spdlog! ***
*** [00:03:42 +02:00] [child 23486] Support for floats 1.23 ***
*** [00:03:42 +02:00] [child 23485] Welcome to spdlog! ***
*** [00:03:42 +02:00] [child 23485] Support for floats 1.23 ***
*** [00:03:42 +02:00] [child 23484] Welcome to spdlog! ***
*** [00:03:42 +02:00] [child 23484] Support for floats 1.23 ***
*** [00:03:42 +02:00] [child 23478] End of example. tid=140392174827392 ***

```

**Comment by gabime** [comment]
I cannot reproduce the pid issue. For example the following works correctly for the pid (but not for the tid!):
```c++
#include "spdlog/spdlog.h"
namespace spd = spdlog;
int main(int, char*[])
{    
        spd::set_pattern("[pid %P] [thread %t] %v");
        // Console logger with color
        auto console = spd::stdout_color_mt("console");
        console->info("Parent");

        auto pid = fork();
        if(pid == 0) {
            console->info("Child");
        }
}
```
and the output is:

```
[pid 18285] [thread 18285] Parent
[pid 18286] [thread 18285] Child
```

**Comment by gabime** [comment]
Fixed the thread ids issue in 799ba2a57bb16b921099bd9ab64a513e4ebe4217 (by providing  option in twekme.h to disable thread id caching)/

**Comment by fcharlie** [comment]
good
